cmd/download.go

```diff
@@ -311,6 +293,51 @@ func (dp downloadPayload) metadata() workspace.ExerciseMetadata {
 	}
 }
 
+type solutionFile struct {
+	path, baseURL, slug string
+}
+
+func (dp downloadPayload) files() []solutionFile {
```

Comment:
I'm wrapping my head around the other types defined here, but for now I would recommend moving up this block so that it is closer to the `downloadPayload` type.

Reply:
Yepp, that's a good call!

I suppose eventually we might want to find a better home for these things, but for now since they're only used by the download command I'm just keeping them in place.